plugins/modules/foreman_architecture.py

```diff
@@ -0,0 +1,126 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_architecture
+short_description: Manage Foreman Architectures using Foreman API
+description:
+  - Create, Update and Delete Foreman Architectures using Foreman API
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+  - "apypie >= 0.0.1"
+options:
+  name:
+    description: Name of architecture
+    required: true
+  operatingsystems:
+    description: List of operating systems the architecture should be assigned to
+    required: false
+    default: None
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: verify ssl connection when communicating with foreman
+    default: true
+    type: bool
+  state:
+    description: Architecture presence
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: "Create an Architecture"
+  foreman_architecture:
+    name: "architecture_name"
+    operatingsystems:
+      - "operatingsystem_name1"
+      - "operatingsystem_name2"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: present
+
+- name: "Update an Architecture"
+  foreman_architecture:
+    name: "architecture_name"
+    operatingsystems:
+      - "operatingsystem_name3"
+      - "operatingsystem_name4"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: present
+
+- name: "Delete an Architecture"
+  foreman_architecture:
+    name: "architecture_name"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'operatingsystems': 'operatingsystem_ids',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            operatingsystems=dict(type='list'),
+            state=dict(default='present', choices=['present', 'absent']),
+        ),
+        supports_check_mode=True,
+    )
+    (entity_dict, state) = module.parse_params()
+
+    module.connect()
+
+    if 'operatingsystems' in entity_dict:
+        entity_dict['operatingsystems'] = module.find_resources('operatingsystems', entity_dict['operatingsystems'], thin=True)
+
+    entity = module.find_resource_by_name('architectures', name=entity_dict['name'], failsafe=True)
+
+    changed = module.ensure_resource_state('architectures', entity_dict, entity, state, name_map)
```

Comment:
```suggestion
    changed = module.ensure_resource_state('architectures', entity_dict, entity, module.state, name_map)
```

see https://github.com/theforeman/foreman-ansible-modules/pull/304